tuf/utils/x509_test.go

```diff
@@ -165,6 +168,19 @@ func TestKeyOperations(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, rsaKey.Private(), decryptedRSAKey.Private())
 
+	// quick test that gun headers are being added appropriately
+	// Encrypt our RSA Key, one type of key should be enough since headers are treated the same
+	testGunKey, err := EncryptPrivateKey(rsaKey, "root", "ilove", "ponies")
```

Comment:
non-blocking nit: while you're in here, can we change `"root"` to `data.CanonicalRootRole` in this file?